### PR TITLE
Added support for more property types.

### PR DESCRIPTION
**Pull request by joshua-davis**
# **Please read the [Contributing guide](CONTRIBUTING.md) before making a PR.**

* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).
---

### Purpose of this PR

Added floatN, halfN, intN, uintN (including float1): https://jira.unity3d.com/browse/PLAT-2229
Fixed Range properties not working, added tests to validate behavior: https://jira.unity3d.com/browse/PLAT-2368
Cleaned up a lot of code that could be consistent across different types to be only based on dimension.

---
### Testing status
Added tests for the majority of the new types and ranges. Tested a sub-set of the new vector property types (e.g. half and half4 but not half2/3).

---
### Comments to reviewers
The majority of the work is in [PropertyDeclarations.cs](https://github.com/Unity-Technologies/Graphics/pull/7463/files#diff-ca6796e3b43a3fc1c9589b92d73cd69240755c61e0e55a8c148262a37daaa613) with most everything else being test related.

### CODE REVIEW

File: TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/EditorTests/PropertyTests/FloatPropertyTests.cs [hunk 1]
```diff
@@ -48,15 +51,120 @@ public IEnumerator FloatProperty_DefaultPropertyValueUsed_IsExpectedColor()
...
             var propBuilder = BuildWithNameOverrides("1");
-            var block = propBuilder.Build(container);
+            var block = propBuilder.Build(container, container._float);
 
             SetupMaterialDelegate materialSetupDelegate = m => { m.SetFloat(propBuilder.PropertyAttribute.UniformName, inputValue); };
             TestSurfaceBlockIsConstantColor(container, propBuilder.BlockName, block, expectedColor, materialSetupDelegate);
             yield break;
         }
+
+        [Test]
```
Comment (jessebarker) [inline TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/EditorTests/PropertyTests/FloatPropertyTests.cs:65]: Yay, new tests!

File: com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/PropertyDeclarations.cs [hunk 2]
```diff
@@ -9,10 +9,25 @@ static class PropertyDeclarations
         {
             {"bool", HandleBoolType },
             {"int", HandleIntegerType },
+            {"int1", HandleIntegerType },
```
Comment (jessebarker) [inline com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/PropertyDeclarations.cs:12]: Don't we need bool vector types here?

File: com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/PropertyDeclarations.cs [hunk 3]
```diff
@@ -420,12 +421,48 @@ static void BuildSimpleProperty(FieldPropertyContext context, string displayType
             MaterialPropertyDeclarationData.BuildSimple(context, displayType, defaultValue, resultProperty);
         }
 
-        static string GetDisplayType(FieldPropertyContext context, string displayType, HashSet<string> allowedDisplayTypeOverrides)
+        static string GetIntegerDisplayType(FieldPropertyContext context)
         {
             foreach (var attribute in context.Attributes)
             {
-                if (allowedDisplayTypeOverrides.Contains(attribute.Name))
+                if (attribute.Name == "Int" || attribute.Name == "Float")
```
Comment (jessebarker) [inline com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/PropertyDeclarations.cs:428]: Is it ok to use string literals here?  Are they guaranteed to be spelled like this?  Could we not see "int" or "float" as well?

File: com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/PropertyDeclarations.cs (hunk 2 above)
Comment (joshua-davis) [reply com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/PropertyDeclarations.cs:12]: We decided no boolN: https://unity.slack.com/archives/C028KHBJG3G/p1651854215902519?thread_ts=1651853526.697259&cid=C028KHBJG3G

**Review by jessebarker — APPROVED** [review]
LGTM

File: com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/PropertyDeclarations.cs (hunk 3 above)
Comment (joshua-davis) [reply com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/PropertyDeclarations.cs:428]: It depends on if we consider our attributes to be case sensitive. Currently they are, so you can say:
`[Property][Int] float MyValue` but not `[int]`. I think this is common for attributes though.
We may consider making this a string in some shared location at some point though.

**Review by aleks01010101 — APPROVED** [review]
LGTM!

**Comment by joshua-davis** [comment]
Yamato tests: https://unity-ci.cds.internal.unity3d.com/job/14441235
The actual foundry tests are green

**Comment by joshua-davis** [comment]
Manually pushed.

**Pull request merged**